library/core/src/ptr/const_ptr.rs

```diff
@@ -1774,3 +1790,146 @@ impl<T: ?Sized> PartialOrd for *const T {
         *self >= *other
     }
 }
+
+#[cfg(kani)]
+#[unstable(feature = "kani", issue = "none")]
+mod verify {
+    use crate::kani;
+
+    #[allow(unused)]
+
+    #[kani::proof_for_contract(<*const i32>::offset)]
+    fn check_offset_slice_i32(){
+        let mut arr: [i32; 5] = kani::any();
+        let test_ptr: *const i32 = arr.as_ptr();
+        let offset: isize = kani::any();
+
+        unsafe{
+            let new_ptr = test_ptr.offset(offset);
+        }
+    }
+
+    #[kani::proof_for_contract(<*const i32>::add)]
+    fn check_add_slice_i32() {
+        let mut arr: [i32; 5] = kani::any();
+        let test_ptr: *const i32 = arr.as_ptr();
+        let count: usize = kani::any();
+        unsafe {
+            let new_ptr = test_ptr.add(count);
+        }
+    }
+
+
+    #[kani::proof_for_contract(<*const i32>::sub)]
+    fn check_sub_slice_i32() {
+        let mut arr: [i32; 5] = kani::any();
+        let test_ptr: *const i32 = arr.as_ptr();
+        let count: usize = kani::any();
+        unsafe {
+            let new_ptr = test_ptr.sub(count);
+        }
+    }
```

Comment:
1. Yes it could be extended to different type, here it is following the minimum requirement in challenge 3.
2. Since the length of array must be defined, I was following the [test 1](https://github.com/model-checking/kani/blob/373af49da5799f73630f6e66e2ff8e96b23a2d92/tests/kani/NondetSlices/test3.rs#L10)
3. kani::any_array() would still need to define the size of array,  by referring to [test2](https://github.com/model-checking/kani/blob/373af49da5799f73630f6e66e2ff8e96b23a2d92/tests/expected/object-bits/insufficient/main.rs#L9). However, I would use it in the next PR.